project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc

```diff
@@ -191,3 +191,54 @@ function _c4m_user_profile_block_user_confirm_email() {
 
   return $block;
 }
+
+/**
+ * Block definition for the user_my_profile_groups block.
+ *
+ * @return array
+ *   The block definition.
+ */
+function _c4m_user_profile_block_user_my_profile_groups() {
+  module_load_include('inc', 'c4m_features_homepage', 'plugins/content_types/my_groups/my_groups');
+  $user = menu_get_object('user', 1);
+  $groups = c4m_features_homepage_get_user_groups_grouped_by_membership($user, array('group'));
+
+  $block = array();
+  $block['subject'] = '';
+  $block['content'] = '';
+  if (is_null($groups)) {
+    return $block;
+  }
+
+  $rendered_groups = '';
+  // Render node views for each group.
+  foreach ($groups as $type => $group_list) {
+    $title = c4m_user_profile_get_group_type_title($type);
+    $rendered_groups .= '<h3>' . $title . '</h3>';
+
+    foreach ($group_list as $group) {
+      $node_view = node_view($group, 'block_list');
+      $rendered_groups .= drupal_render($node_view);
+    }
+  }
+
+  $variables = array(
+    'groups' => $rendered_groups,
+  );
+
+  $block['content'] = $rendered_groups ? theme('c4m_user_profile_my_profile_groups', $variables) : FALSE;
```

Comment:
@DavidHernandez 
It seems that at this point `$rendered_groups` will always returns TRUE ...